From 07b16575d096a509be6f2edc6143e8d3122b3af9 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Tue, 11 Sep 2018 17:25:42 +0800
Subject: [PATCH 4581/5242] MLK-19544 media: mipi_csi: check camera's running
 state before suspend

commit  9fb8c1abf24de17e05ac00babe40a14faaa00c03 from
https://source.codeaurora.org/external/imx/linux-imx.git

don't support suspend when camera's running

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit 8a399818a56d669b8b862e3c6c1718199c81854e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/mxc/capture/mxc_mipi_csi.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
index f9346f1..b4b15e5 100644
--- a/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
+++ b/drivers/media/platform/mxc/capture/mxc_mipi_csi.c
@@ -1237,6 +1237,14 @@ static int mipi_csis_pm_resume(struct device *dev, bool runtime)
 #ifdef CONFIG_PM_SLEEP
 static int mipi_csis_suspend(struct device *dev)
 {
+	struct platform_device *pdev = to_platform_device(dev);
+	struct csi_state *state = platform_get_drvdata(pdev);
+
+	if (state->flags & ST_STREAMING) {
+		dev_warn(dev, "running, prevent entering suspend.\n");
+		return -EAGAIN;
+	}
+
 	return mipi_csis_pm_suspend(dev, false);
 }
 
-- 
1.7.9.5

